### ISSUE TITLE
convert from ps file shifts and crops the content per page

### ISSUE THREAD

**Issue by pgajdos** [ImageMagick/ImageMagick#378]
Tested with 6.9.7-7.

convert Bezier_M1_M2.ps <pdf or jpg>

Testcase:
[Bezier_M1_M2.ps.gz](https://github.com/ImageMagick/ImageMagick/files/771378/Bezier_M1_M2.ps.gz)

see https://bugzilla.suse.com/show_bug.cgi?id=1023454

Tested also with ps2pdf, this does not exhibit the problem.

**Comment by dlemstra** [comment]
Which Ghostscript version are you using? And did you confirm that this is not a Ghostscript bug?

**Comment by pgajdos** [comment]
ghostscript is of version 9.20. I do not know how to check that it is a ghostscript bug besides trying ps2pdf, as I already did, see my initial comment.

**Comment by dlemstra** [comment]
I was not aware that ps2pdf was a Ghostscript component. We will take a look later this week. It might be that you will will need to use `-define eps:use-cropbox=true` in your convert command. Can you give that a try?

**Comment by pgajdos** [comment]
Aha, ps2pdf seems it is shell script on my system, frontend to gs.

From the initial comment, second parameter was stripped (contained less than and greather than symbols). Neither following to the correct job (see second and third page):

`convert Bezier_M1_M2.ps -define eps:use-cropbox=true Bezier_M1_M2.pdf`
`convert Bezier_M1_M2.ps -define eps:use-cropbox=true Bezier_M1_M2.jpg`

**Comment by pgajdos** [comment]
Hey,

JFYI, tested with 7.0.5-4, still the same result, even with `-define eps:use-cropbox=`.

**Comment by pgajdos** [comment]
Tested again with 7.0.7-28, still the same result.

**Comment by dlemstra** [comment]
We automatically set the `-g` option when we read a file with the `ps` coder. This does not happen with the `pdf` coder. I don't understand why this option would break this but I can get the output that you want with the following command:
`convert PDF:Bezier_M1_M2.ps  Bezier_M1_M2.jpg`.

**Comment by pgajdos** [comment]
The original problem, created by `convert Bezier_M1_M2.ps Bezier_M1_M2.pdf`
(A)
![result](https://user-images.githubusercontent.com/4067843/49141944-0a18aa80-f2f8-11e8-81b4-d4f545987a86.png)

Ordinary command `gs -sDEVICE=jpeg -o out-%d.jpg Bezier_M1_M2.ps` produces expected
(B)
![result](https://user-images.githubusercontent.com/4067843/49142562-587a7900-f2f9-11e8-8adc-50e7bf0d287b.png)

Command which is constructed by ImageMagick seems to be:
```
(gdb) p command
$4 = 0x7ffffffee490 "'gs' -sstdout=%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT 
-dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 '-sDEVICE=pngalpha' -dTextAlphaBits=4
-dGraphicsAlphaBits=4 '-r72x72' -g504x720"...
```

Indeed, with
```
gs -sstdout=%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT \
   -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 '-sDEVICE=pngalpha' -o out-%d.png \
   -dTextAlphaBits=4 -dGraphicsAlphaBits=4 '-r72x72' -g504x720 Bezier_M1_M2.ps
```
I get
(C)
![result](https://user-images.githubusercontent.com/4067843/49143112-9deb7600-f2fa-11e8-8712-c9de73cd737f.png)

With `convert PDF:Bezier_M1_M2.ps Bezier_M1_M2.jpg` I get expected output
(D)
![result](https://user-images.githubusercontent.com/4067843/49143953-77c6d580-f2fc-11e8-826f-e977bbfe34c4.png)

To sum up:
* (D) produces similar wanted result as (B), it seems to be shifted at all and seem to have more space on margins.
* (C) does not produce wanted output, all pages are shifted, including the first one unlike (A).
* removing `-g504x720` indeed fixes the issue and produces wanted output, something like in (D).

From my not-very-familiar point of view it seems that convert wants to reduce margin size, but only shifts the first page. Adding Johannes (@jsmeix) for more insight.

**Comment by jsmeix** [comment]
I am afraid I am not at all a sufficient Ghostscript
or PostScript or PDF expert to provide useful help here.
I suggest to ask at Ghostscript upstream.
See https://ghostscript.com/faq.html
for possible ways how to ask at Ghostscript upstream
e.g. see the items
`Where can I get more information ...`
and
`Are there any specific discussion forums ...`
